api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/HivCohortQueries.java

```diff
@@ -191,6 +197,175 @@ public CohortDefinition getPatientsInArtWhoAbandoned() {
+            + "               inner join encounter e "
+            + "                       ON e.patient_id = p.patient_id "
+            + "               inner join obs o "
+            + "                       ON o.encounter_id = e.encounter_id "
+            + "        WHERE  p.voided = 0 "
+            + "               AND e.voided = 0 "
+            + "               AND o.voided = 0 "
+            + "               AND o.concept_id = %s "
+            + "               AND o.value_datetime IS NOT NULL "
+            + "               AND e.encounter_type = %s) most_recent "
+            + "WHERE  most_recent.value_datetime = (SELECT Max(value_datetime) "
+            + "                                     FROM   obs "
+            + "                                     WHERE  encounter_id = "
+            + "                                            most_recent.encounter_id) "
+            + "       AND most_recent.value_datetime < Date_add(:onOrBefore, interval 30 day) "
+            + "       AND most_recent.location_id = :location "
+            + "GROUP  BY most_recent.patient_id ";
+
+    definition.setQuery(
+        String.format(
+            query,
+            hivMetadata.getARVPharmaciaEncounterType().getEncounterTypeId(),
+            hivMetadata.getReturnVisitDateForArvDrugConcept().getConceptId(),
+            hivMetadata.getAdultoSeguimentoEncounterType().getEncounterTypeId(),
+            hivMetadata.getARVPediatriaSeguimentoEncounterType().getEncounterTypeId(),
+            commonMetadata.getReturnVisitDateConcept().getConceptId(),
+            hivMetadata.getArtDatePickup().getConceptId(),
+            hivMetadata.getMasterCardDrugPickupEncounterType().getEncounterTypeId()));
+
+    return definition;
+  }
+
+  /**
+   * 14. All patients who do not have the next scheduled drug pick up date (Fila) and next scheduled
+   * consultation date (Ficha de Seguimento or Ficha Clinica – Master Card) and ART Pickup date
+   * (Recepção – Levantou ARV).
+   *
+   * @return
+   */
+  @DocumentedDefinition(value = "patientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup")
+  private CohortDefinition getPatientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup() {
+    SqlCohortDefinition definition = new SqlCohortDefinition();
+
+    definition.setName("patientWithoutScheduledDrugPickupDateMasterCardAmdArtPickup");
+
+    definition.addParameter(new Parameter("location", "location", Location.class));
+
+    String query =
+        "SELECT p.patient_id "
+            + "FROM   patient p "
+            + "       inner join encounter e "
+            + "               ON e.patient_id = p.patient_id "
+            + "       inner join obs o "
+            + "               ON o.encounter_id = e.encounter_id "
+            + "WHERE  p.voided = 0 "
+            + "       AND e.voided = 0 "
+            + "       AND o.voided = 0 "
+            + "       AND e.encounter_type NOT IN ( %s, %s, %s, %s ) "
```

Comment:
@ynurmahomed this query will not have the same effect of filtering out patients without these encounters.

For example in cases where patients have other encounters outside of this list, they will still be excluded. This is the same issue that @cfaife had on TxCurr. Pls speak to him or we can have a call if you have any doubts about this

Reply:
wait sorry I am getting confused, this is not exactly the same scenario as txCurr, pls ignore :)

Reply:
So for example if a patient has encounters that are not in this list, then this query will pick them, even though they have encounters that are in fact in this list

Reply:
Re-using from TX_CURR.